### ISSUE TITLE
Sentence transformer的sample 代码与GGUF格式的计算结果不同

### ISSUE THREAD

**Issue by zcuder** [QwenLM/Qwen3-Embedding#91]
Sentence transformer的sample code from https://huggingface.co/Qwen/Qwen3-Embedding-8B
GGUF https://huggingface.co/Qwen/Qwen3-Embedding-8B-GGUF
代码引用自https://huggingface.co/Qwen/Qwen3-Embedding-8B

`from sentence_transformers import SentenceTransformer

model = SentenceTransformer("Qwen/Qwen3-Embedding-8B")

queries = [
"What is the capital of China?",
"Explain gravity",
]
documents = [
"The capital of China is Beijing.",
"Gravity is a force that attracts two bodies towards each other. It gives weight to physical objects and is responsible for the movement of planets around the sun.",
]

query_embeddings = model.encode(queries, prompt_name="query")
document_embeddings = model.encode(documents)

similarity = model.similarity(query_embeddings, document_embeddings)
print(similarity)
`

这个结果出来[[0.7493, 0.0751] [0.0880, 0.6318]]

如果用https://huggingface.co/Qwen/Qwen3-Embedding-8B-GGUF/tree/main 上下载的[Qwen3-Embedding-8B-Q8_0.gguf](https://huggingface.co/Qwen/Qwen3-Embedding-8B-GGUF/blob/main/Qwen3-Embedding-8B-Q8_0.gguf)
喂入ollama之后，
`
import ollama
import numpy as np
import torch
import torch.nn.functional as F # 关键导入！

def get_detailed_instruct(task_description: str, query: str) -> str:
return f'Instruct: {task_description}\nQuery:{query}'

task = 'Given a web search query, retrieve relevant passages that answer the query'
queries = [
get_detailed_instruct(task, 'What is the capital of China?'),
get_detailed_instruct(task, 'Explain gravity')
]
documents = [
"The capital of China is Beijing.",
"Gravity is a force that attracts two bodies towards each other..."
]
input_texts = queries + documents

embeddings = []
for text in input_texts:
response = ollama.embeddings(model='qwen3-embedding-8B-q5_k_m:latest', prompt=text) # 或 model='qwen:7b'
embeddings.append(response["embedding"])
embeddings_tensor = torch.tensor(embeddings, dtype=torch.float32)
embeddings_normalized = F.normalize(embeddings_tensor, p=2, dim=1)
embeddings = embeddings_normalized.numpy()
scores = embeddings[:2] @ embeddings[2:].T

print(scores.tolist())
`
结果
[[0.7415439486503601, 0.2103012204170227], [0.17427761852741241, 0.467174232006073]]

有意思的是
我试着自己用convert_hf_to_gguf.py把hf模型导成gguf再导入ollama之后，同样用以上脚本计算
[[0.7617605924606323, 0.09762856364250183], [0.08815209567546844, 0.6415960788726807]]
这个结果跟sentence transformer接近，所以[gguf](https://huggingface.co/Qwen/Qwen3-Embedding-8B-GGUF) 的模型是做了什么特殊处理吗？为什么跟sentence transformer差别比较大？
